custom-login/src/helpers.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * Helper function that watches the authenticate state, then applies it
+ * as a boolean (authenticated) as well as attaches the userinfo data.
+ */
+async function checkAuthentication() {
+  const authenticated = await this.props.auth.isAuthenticated();
+  if (authenticated !== this.state.authenticated) {
+    if (authenticated && !this.state.userinfo) {
+      const userinfo = await this.props.auth.getUser();
+      this.setState({ authenticated, userinfo });
```

Comment:
this is a bit weird, but I understand why we don't want to copy this logic everywhere

Reply:
The helper file is weird?  It does feel all alone, any other ideas?